File: mobile-wallet-master/core/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/UserEntityMapper.java
package org.mifos.mobilewallet.core.data.fineract.entity.mapper;

import org.mifos.mobilewallet.core.data.fineract.entity.UserEntity;
import org.mifos.mobilewallet.core.domain.model.user.User;

import javax.inject.Inject;

/**
 * Created by naman on 17/6/17.
 */

public class UserEntityMapper {

    @Inject
    public UserEntityMapper() {
    }

    public User transform(UserEntity userEntity) {
        User user = new User();

        if (userEntity != null) {
            user.setUserId(userEntity.getUserId());
            user.setUserName(userEntity.getUserName());
            user.setAuthenticationKey(userEntity.getBase64EncodedAuthenticationKey());

        }
        return user;
    }
}

File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/UserEntityMapper.kt
package org.mifos.mobilewallet.core.data.fineract.entity.mapper

import com.mifos.mobilewallet.model.domain.user.User
import com.mifos.mobilewallet.model.entity.UserEntity
import javax.inject.Inject

/**
 * Created by naman on 17/6/17.
 */
class UserEntityMapper @Inject constructor() {
    fun transform(userEntity: UserEntity?): User {
        val user = User()
        if (userEntity != null) {
            user.userId = userEntity.userId
            user.userName = userEntity.userName
            user.authenticationKey = userEntity.base64EncodedAuthenticationKey
        }
        return user
    }
}
